### PR TITLE
Update cats, kind-projector. Remove silencer-plugin.

### PR DESCRIPTION
**Pull request by kellydavid**
- Updated cats and cats-effect to 2.3.0
 - Updated kind-projector to 0.11.2
 - Removed silencer plugin (not required for scala 2.13)
 - Switched from `import scala.collection.JavaConverters._` to `import scala.jdk.CollectionConverters._`
 - Formatting

### CODE REVIEW

File: src/main/scala/com/tenable/library/kafkaclient/config/KafkaProducerConfig.scala [hunk 1]
```diff
@@ -76,15 +79,18 @@ object KafkaProducerConfig {
...
     )
   }
 
   def default(connectionString: String): KafkaProducerConfig =
     new KafkaProducerConfig(
       connectionString,
       ackStrategy = AckStrategy.One,
-      retries = 0,                     //if not configured with max batch in flight = 1, retries can mess up ordering.  disable by default for now
+      retries =
+        0,                             //if not configured with max batch in flight = 1, retries can mess up ordering.  disable by default for now
```
Comment (niqdev) [inline src/main/scala/com/tenable/library/kafkaclient/config/KafkaProducerConfig.scala:93]: this formatting is a bit ugly 😓

File: src/main/scala/com/tenable/library/kafkaclient/client/standard/KafkaProducerIO.scala [hunk 2]
```diff
@@ -39,8 +38,7 @@ trait KafkaProducerIO[F[_], K, V] { self =>
       decorate: ProducerRecord[K, V] => ProducerRecord[K, V]
   ): F[List[RecordMetadata]]
 
-  /**
-    * When sending an event using the asynchronous Kafka client, there are two stages of `done`:
+  /** When sending an event using the asynchronous Kafka client, there are two stages of `done`:
```
Comment (niqdev) [inline src/main/scala/com/tenable/library/kafkaclient/client/standard/KafkaProducerIO.scala:41]: if you want to keep the same formatting this should work
```
# https://scalameta.org/scalafmt/docs/configuration.html#docstrings
docstrings.blankFirstLine = yes
docstrings.style = SpaceAsterisk
```

**Review by niqdev — APPROVED** [review]
LGTM

File: src/main/scala/com/tenable/library/kafkaclient/client/standard/KafkaProducerIO.scala (hunk 2 above)
Comment (kellydavid) [reply src/main/scala/com/tenable/library/kafkaclient/client/standard/KafkaProducerIO.scala:41]: Thanks! I think I might propose a different formatting in another PR...

**Pull request merged**